org.eclipse.jdt.ls.debug/src/org/eclipse/jdt/ls/debug/adapter/ProtocolServer.java

```diff
@@ -0,0 +1,229 @@
+/*******************************************************************************
+* Copyright (c) 2017 Microsoft Corporation and others.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Eclipse Public License v1.0
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v10.html
+*
+* Contributors:
+*     Microsoft Corporation - initial API and implementation
+*******************************************************************************/
+
+package org.eclipse.jdt.ls.debug.adapter;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.io.Writer;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.eclipse.jdt.ls.debug.internal.Logger;
+
+public class ProtocolServer {
+    private static final int BUFFER_SIZE = 4096;
+    private static final String TWO_CRLF = "\r\n\r\n";
```

Comment:
yes, i tried it on Mac, it also uses hard coded "\r\n\r\n" as the separator of header and body for request and response message.